car.go

```diff
@@ -159,7 +171,25 @@ func (cr *CarReader) Next() (blocks.Block, error) {
 		return nil, fmt.Errorf("mismatch in content integrity, name: %s, data: %s", c, hashed)
 	}
 
-	return blocks.NewBlockWithCid(data, c)
+	offset := cr.offset + l - uint64(len(data))
+	cr.offset += l
+
+	blk, err := blocks.NewBlockWithCid(data, c)
+	if cr.fullPath == "" {
+		return blk, err
+	}
+	nd, err := format.Decode(blk)
```

Comment:
From what I can tell this is going to force a full decode of the data using the appropriate codec, which is a new piece of functionality for go-car and possibly a significant new cost. I don't know why `Filestore` needs a full `Node` since it's only touching `RawData()` and `Cid()`, the elements of `Block`, so we're being forced here to decode a full Node just because the Filestore API wants the whole-hog?

This seems like a significant cost overhead to me, just for the use-case of being able to write to a Filestore which may not be a majority use-case. Am I right? Got any thoughts on how we might make this more efficient for the common use-case or even wipe away this decode entirely?